docs/dev-guide.md

```diff
@@ -458,48 +458,6 @@ sqldb.queryOneRow(sql.block_name, params, function (err, result) {
 });
 ```
 
-- For transactions with correct error handling use following pattern. It is important to use `async.series()` to run all the operations rather than using a callback stack, because with `async.series()` we can guarantee that `endTransaction()` is called no matter whether any of the intermediate operations produce an error.
-
-```javascript
-// do this
-sqldb.beginTransaction(function (err, client, done) {
-  if (ERR(err, callback)) return;
-  async.series(
-    [
-      function (callback) {
-        // only use queryWithClient() and queryWithClientOneRow() inside the transaction
-        sqldb.queryWithClient(client, sql.block_name, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          // do things
-          callback(null);
-        });
-      },
-      // more series functions inside the transaction
-    ],
-    function (err) {
-      sqldb.endTransaction(client, done, err, function (err) {
-        // will rollback if err is defined
-        if (ERR(err, callback)) return;
-        // transaction successfully committed at this point
-        callback(null);
-      });
-    }
-  );
-});
-
-// don't do this
-sqldb.beginTransaction(function (err, client, done) {
-  if (ERR(err, callback)) return;
-  sqldb.queryWithClient(client, sql.block_name, params, function (err, result) {
-    if (ERR(err, callback)) return; // THIS IS WRONG, it may exit without endTransaction()
-    sqldb.endTransaction(client, done, err, function (err) {
-      if (ERR(err, callback)) return;
-      callback(null);
-    });
-  });
-});
-```
-
```

Comment:
Should we replace this with documentation of how to use `runInTransactionAsync()`, or are we moving to a documentation model where we'll document this in the `sqldb` package and expect developers to find it there?

Reply:
Over on https://github.com/PrairieLearn/PrairieLearn/pull/7003#discussion_r1089378055 I brought up basically the same question. I do think there's space for an overarching dev guide (although I think that should be multiple pages and not one monolithic document), but I don't know that `docs/dev-guide.md` will be a useful starting point for that. I'm leaning towards just deleting for now and worrying about improved dev docs later, as I don't think this giant doc is particularly useful in its current form.

Reply:
I agree the giant doc is not very useful. Per-package docs seem like the highest priority to me (as we pull `sqldb` and similar things out into standalone packages). But this can wait.